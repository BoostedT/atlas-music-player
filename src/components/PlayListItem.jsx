import React from "react";

export default function PlayListItem({ title, artist, length, active }) {
  return (
    <div
      className={`
        group flex items-center justify-between rounded-lg px-3 py-2 cursor-pointer transition-all
        ${active 
          ? "bg-jelly-500/20 text-cloud-50 shadow-md" 
          : "hover:bg-night-800 hover:scale-[1.02] hover:shadow-md text-smoke-400"}
      `}
    >
      {/* Left: Title + Artist */}
      <div>
        <div className={`font-semibold ${active ? "text-jelly-400" : "group-hover:text-jelly-400"}`}>
          {title}
        </div>
        <div className="text-sm text-mist-300 group-hover:text-mist-100">
          {artist}
        </div>
      </div>

      {/* Right: Song length */}
      <span
        className={`ml-2 text-sm ${active ? "text-jelly-400" : "group-hover:text-cloud-50"}`}
      >
        {length}
      </span>
    </div>
  );
}
