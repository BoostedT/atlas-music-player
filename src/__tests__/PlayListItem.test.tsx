import { render, fireEvent } from "@testing-library/react";
import PlayListItem from "../components/PlayListItem.tsx";
import React from "react";

const baseProps = {
  title: "Test Song",
  artist: "Test Artist",
  duration: 200,
  active: false,
  onClick: vi.fn(),
};

describe("PlayListItem", () => {
  it("renders inactive playlist item", () => {
    const { asFragment } = render(<PlayListItem {...baseProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders active playlist item", () => {
    const { asFragment } = render(<PlayListItem {...baseProps} active={true} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("calls onClick when clicked", () => {
    const mockClick = vi.fn();
    const { getByRole } = render(<PlayListItem {...baseProps} onClick={mockClick} />);
    fireEvent.click(getByRole("button"));
    expect(mockClick).toHaveBeenCalled();
  });
});
