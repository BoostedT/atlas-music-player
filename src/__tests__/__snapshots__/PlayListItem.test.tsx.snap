// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PlayListItem > renders active playlist item 1`] = `
<DocumentFragment>
  <button
    class="
        group flex items-center justify-between rounded-lg px-3 py-2 transition-all text-left w-full
        bg-jelly-500/20 text-cloud-50 shadow-md
      "
    type="button"
  >
    <div>
      <div
        class="font-semibold text-jelly-400"
      >
        Test Song
      </div>
      <div
        class="text-sm text-mist-300 group-hover:text-mist-100"
      >
        Test Artist
      </div>
    </div>
    <span
      class="ml-2 text-sm text-jelly-400"
    >
      3:20
    </span>
  </button>
</DocumentFragment>
`;

exports[`PlayListItem > renders inactive playlist item 1`] = `
<DocumentFragment>
  <button
    class="
        group flex items-center justify-between rounded-lg px-3 py-2 transition-all text-left w-full
        hover:bg-night-800 hover:scale-[1.02] hover:shadow-md text-smoke-400
      "
    type="button"
  >
    <div>
      <div
        class="font-semibold group-hover:text-jelly-400"
      >
        Test Song
      </div>
      <div
        class="text-sm text-mist-300 group-hover:text-mist-100"
      >
        Test Artist
      </div>
    </div>
    <span
      class="ml-2 text-sm group-hover:text-cloud-50"
    >
      3:20
    </span>
  </button>
</DocumentFragment>
`;
